packages/central-server/src/apiV2/entityRelations/assertEntityRelationPermissions.js

```diff
@@ -0,0 +1,54 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2023 Beyond Essential Systems Pty Ltd
+ */
+
+import { TYPES } from '@tupaia/database';
+import { hasBESAdminAccess } from '../../permissions';
+import { mergeFilter, mergeMultiJoin } from '../utilities';
+
+export const assertEntityRelationPermissions = async (accessPolicy, models, entityRelationId) => {
+  const entityRelation = await models.entityRelation.findById(entityRelationId);
+  if (!entityRelation) {
+    throw new Error(`No entity relation exists with id ${entityRelationId}`);
+  }
+
+  const child = await models.entity.findById(entityRelation.child_id);
+  if (!accessPolicy.allows(child.country_code)) {
+    throw new Error('You do not have permissions for this entity relation');
+  }
+
+  return true;
+};
+
+export const createEntityRelationDbFilter = (accessPolicy, criteria, options) => {
+  const dbConditions = { ...criteria };
+  const dbOptions = { ...options };
+
+  if (!hasBESAdminAccess(accessPolicy)) {
+    // Our permissions check logic is simplified based on the following assumptions:
```

Comment:
Hopefully this makes sense? The alternative would be to use a quite convoluted RAW criteria logic similar to the one in [assertDashboardRelationsPermissions.js#L141](https://github.com/beyondessential/tupaia/blob/dev/packages/central-server/src/apiV2/dashboardRelations/assertDashboardRelationsPermissions.js#L141), that checks permissions to the associated country (non-project) / any associated coumntry (project).

What's more, this should be done for both `child_id` and `parent_id`.